source/EffectfulForcing/MFPSAndVariations/ContinuityProperties.lagda

```diff
@@ -0,0 +1,913 @@
+
+  IH₂ : α ＝⟪ ns ⟫₀ β
+  IH₂ = ＝⟦⟧-implies-＝⟪⟫₀ α β (φ ₁) (ψ ₁)
+
+  † : α ＝⟪ ms ++ ns ⟫₀ β
+  † = ＝⟪⟫-++-lemma₂ α β ms ns (IH₁ , IH₂)
+
+\end{code}
+
+\begin{code}
+
+to-bool-congruence : (m n : ℕ)
+                   → (𝒷₁ : is-boolean-valued m)
+                   → (𝒷₂ : is-boolean-valued n)
+                   → m ＝ n
+                   → to-bool m 𝒷₁ ＝ to-bool n 𝒷₂
+to-bool-congruence zero            zero            (inl refl) (inl refl) _ = refl
+to-bool-congruence (succ zero)     (succ zero)     (inr refl) (inr refl) _ = refl
+to-bool-congruence (succ (succ _)) (succ (succ _)) (inl ())   (inl _)    _
+to-bool-congruence (succ (succ _)) (succ (succ _)) (inl ())   (inr _)    _
+to-bool-congruence (succ (succ _)) (succ (succ _)) (inr ())   (inl _)    _
+to-bool-congruence (succ (succ _)) (succ (succ _)) (inr ())   (inr _)    _
+
+\end{code}
+
+\begin{code}
+
+to-cantor-＝⟦⟧ : {α₁ α₂ : Baire}
+                 (ϑ₁ : is-boolean-point α₁) (ϑ₂ : is-boolean-point α₂)
+                 (t : BT ℕ)
+               → α₁ ＝⟦ t ⟧ α₂
+               → to-cantor (α₁ , ϑ₁) ＝⟦ t ⟧ to-cantor (α₂ , ϑ₂)
+to-cantor-＝⟦⟧ {α₁} {α₂} ϑ₁ ϑ₂ []       _      = []
+to-cantor-＝⟦⟧ {α₁} {α₂} ϑ₁ ϑ₂ (n ∷ φ) (p ∷ ψ) = β ∷ γ
+ where
+  β : to-bool (α₁ n) (ϑ₁ n) ＝ to-bool (α₂ n) (ϑ₂ n)
+  β = to-bool-congruence (α₁ n) (α₂ n) (ϑ₁ n) (ϑ₂ n) p
+
+  γ : (b : 𝟚) → to-cantor (α₁ , ϑ₁) ＝⟦ φ b ⟧ to-cantor (α₂ , ϑ₂)
+  γ ₀ = to-cantor-＝⟦⟧ ϑ₁ ϑ₂ (φ ₀) (ψ ₀)
+  γ ₁ = to-cantor-＝⟦⟧ ϑ₁ ϑ₂ (φ ₁) (ψ ₁)
+
+\end{code}
+
+At this point, we are ready to prove one direction of the equivalence we are
+interested in: `is-uniformly-continuous` implies `is-uniformly-continuous₀`.
+
+\begin{code}
+
+uni-continuity-implies-uni-continuity₀ : (f : Cantor → ℕ)
+                                       → is-uniformly-continuous  f
+                                       → is-uniformly-continuous₀ f
+uni-continuity-implies-uni-continuity₀ f 𝔠 =
+ n , λ (α₁ , ϑ₁) (α₂ , ϑ₂) → † α₁ α₂ ϑ₁ ϑ₂
+  where
+   t : BT ℕ
+   t = pr₁ 𝔠
+
+   n : ℕ
+   n = succ (maximumᵤ (pr₁ 𝔠))
+
+   f₀ : Cantor₀ → ℕ
+   f₀ = to-cantor₀-map f
+
+   † : (α₁ α₂ : Baire) (ϑ₁ : is-boolean-point α₁) (ϑ₂ : is-boolean-point α₂)
+     → α₁ ＝⦅ n ⦆ α₂ → f₀ (α₁ , ϑ₁) ＝ f₀ (α₂ , ϑ₂)
+   † α₁ α₂ ϑ₁ ϑ₂ (p , q) = pr₂ 𝔠 (to-cantor (α₁ , ϑ₁)) (to-cantor (α₂ , ϑ₂)) η
+    where
+     ϝ : tl α₁ ＝⦅ maximumᵤ′ t ⦆ (tl α₂)
+     ϝ = transport
+          (λ - → tl α₁ ＝⦅ - ⦆ tl α₂)
+          (maximumᵤ′-equivalent-to-maximumᵤ t)
+          q
+
+     χ : α₁ ＝⦅ succ (maximum (sequentialize t)) ⦆ α₂
+     χ = p , ϝ
+
+     υ : α₁ ＝⟪ sequentialize t ⟫ α₂
+     υ = ＝⦅⦆-implies-＝⟪⟫ α₁ α₂ (sequentialize t) χ
+
+     ξ : α₁ ＝⟪ sequentialize t ⟫₀ α₂
+     ξ = ＝⟪⟫-implies-＝⟪⟫₀ α₁ α₂ (sequentialize t) υ
+
+     ζ : α₁ ＝⟦ t ⟧ α₂
+     ζ = ＝⟪⟫₀-implies-＝⟦⟧ α₁ α₂ t ξ
+
+     η : to-cantor (α₁ , ϑ₁) ＝⟦ t ⟧ to-cantor (α₂ , ϑ₂)
+     η = to-cantor-＝⟦⟧ ϑ₁ ϑ₂ t ζ
+
+\end{code}
+
+To prove the converse direction, we define an analogue of the range function
```

Comment:
`To prove the converse direction,` -> `To prove the converse,`.

Reply:
Fixed.